pennylane/ops/op_math/exp.py

```diff
@@ -245,19 +285,27 @@ def resource_params(self) -> dict:
 
     # pylint: disable=invalid-overridden-method, arguments-renamed
     @property
+    @queuing.QueuingManager.stop_recording()
     def has_decomposition(self):
         # TODO: Support nested sums in method
-        base = self.base
+        base = self.base.simplify()
         coeff = self.coeff
-        if isinstance(base, SProd):
+        while isinstance(base, SProd):
```

Comment:
doesn't the `simplify()` already flatten out any nested structures?

Reply:
was trying to mimic the recursive nature of of `_recursive_definition`, but should just be able to do one layer of unwrapping now.